### PR TITLE
PICT-207: Cancel Refund via Refunds API

### PR DESCRIPTION
**Pull request by NghiaDTr**

### CODE REVIEW

**Review by Tung-Huynh-Shopmacher — CHANGES_REQUESTED** [review]

File: processor/src/mollie/payment.mollie.ts [hunk 1]
```diff
@@ -17,7 +19,19 @@ import CustomError from '../errors/custom.error';
  * @return {Promise<Payment>} A promise that resolves to the created Payment object.
  */
 export const createMolliePayment = async (paymentParams: PaymentCreateParams): Promise<Payment> => {
-  return await initMollieClient().payments.create(paymentParams);
+  try {
+    return await initMollieClient().payments.create(paymentParams);
+  } catch (error: unknown) {
+    let errorMessage;
+
+    if (error instanceof MollieApiError) {
```
Comment (Tung-Huynh-Shopmacher) [inline processor/src/mollie/payment.mollie.ts:27]: add console.error

**Review by Tung-Huynh-Shopmacher — CHANGES_REQUESTED** [review]

File: docs/CreatePaymentWithSpecificParams.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+| `paymentMethodInfo.method: "creditcard"`                                                                    | `method: creditcard`                           | YES      |
+| `custom.fields.sctm_create_payment_request.cardToken: "card_token_12345"`                                                                    |                         | YES      |
+| `amountPlanned.currencyCode: "EUR"`                                                                       | `amount.currency: EUR`                       | YES      |
+| `amountPlanned.centAmount: "1000"` and `amountPlanned.fractionDigits: "2"`                                | `amount.value: "10.00"`                      | YES      |
+| `custom.fields.sctm_create_payment_request.redirectUrl: "https://webshop.example.org/order/12345/"`       | `redirectUrl: "https://webshop.example.org/order/12345/"`                      | YES      |
+| `custom.fields.sctm_create_payment_request.webhookUrl: "https://webshop.example.org/payments/webhook/"`   | `redirectUrl: "https://webshop.example.org/payments/webhook/"`                      | NO      |
```
[the remark this answers is not in the document]
Comment (NghiaDTr) [reply docs/CreatePaymentWithSpecificParams.md]: Yes I removed it

File: processor/src/mollie/payment.mollie.ts [hunk 3]
```diff
@@ -47,3 +61,20 @@ export const listPaymentMethods = async (options: MethodsListParams): Promise<Li
...
+    return await initMollieClient().paymentRefunds.cancel(refundId, params);
+  } catch (error: unknown) {
+    let errorMessage;
+    if (error instanceof MollieApiError) {
+      errorMessage = `SCTM - cancelPaymentRefund - error: ${error.message}`;
+    } else {
+      errorMessage = 'SCTM - cancelPaymentRefund - Failed to cancel the refund with unknown errors';
+    }
+
+    logger.error(errorMessage);
```
Comment (Tung-Huynh-Shopmacher) [inline processor/src/mollie/payment.mollie.ts:76]: add prefix for the error, where it happened "Cancel payment refund"?

**Review by Tung-Huynh-Shopmacher — CHANGES_REQUESTED** [review]

File: processor/src/mollie/payment.mollie.ts [hunk 4]
```diff
@@ -17,7 +19,21 @@ import CustomError from '../errors/custom.error';
...
-  return await initMollieClient().payments.create(paymentParams);
+  try {
+    return await initMollieClient().payments.create(paymentParams);
+  } catch (error: unknown) {
+    let errorMessage;
+
+    if (error instanceof MollieApiError) {
+      errorMessage = `SCTM - createMolliePayment - error: ${error.message}, field: ${error.field}`;
+    } else {
+      errorMessage = 'SCTM - createMolliePayment - Failed to create payment with unknown errors';
```
Comment (Tung-Huynh-Shopmacher) [inline processor/src/mollie/payment.mollie.ts:30]: logger.error unknown error for debug

File: processor/src/mollie/payment.mollie.ts [hunk 5]
```diff
@@ -47,3 +63,20 @@ export const listPaymentMethods = async (options: MethodsListParams): Promise<Li
...
+
+export const cancelPaymentRefund = async (refundId: string, params: CancelParameters): Promise<boolean> => {
+  try {
+    return await initMollieClient().paymentRefunds.cancel(refundId, params);
+  } catch (error: unknown) {
+    let errorMessage;
+    if (error instanceof MollieApiError) {
+      errorMessage = `SCTM - cancelPaymentRefund - error: ${error.message}`;
+    } else {
+      errorMessage = 'SCTM - cancelPaymentRefund - Failed to cancel the refund with unknown errors';
```
Comment (Tung-Huynh-Shopmacher) [inline processor/src/mollie/payment.mollie.ts:75]: logger.error unknown error for debug

**Review by Tung-Huynh-Shopmacher — CHANGES_REQUESTED** [review]

**Pull request merged**